src/server/render.ts

```diff
@@ -137,6 +139,19 @@ function defaultCsp() {
   };
 }
 
+function getSyncPluginsMessage(plugins: Plugin[]) {
+  return `Async server components cannot be rendered synchronously. The following plugins use a synchronous render method: "${
+    plugins.map((plugin) => plugin.name).join('", "')
+  }"`;
+}
+
+function checkAsyncComponent(
+  component: unknown,
+): component is AsyncRoute | AsyncLayout {
+  return typeof component === "function" &&
+    component.constructor.name === "AsyncFunction";
```

Comment:
```suggestion
  return component.constructor.name === "AsyncFunction";
```

Reply:
`route.component` can be `undefined` in the case a route only has handlers.